client/src/components/Auth/AuthProvider.js

```diff
@@ -20,28 +21,24 @@ const AuthProvider = ({ children }) => {
             callback()
         } catch (err) {
             console.log(`Login error ${err}`)
-
-            // Assignment: what should we do if this fails?
-            //Alert the user and re route to login page 
+            navigate('/login')
             alert('password or email wrong')
-
         }
 
     }
 
     const register = async(email, password, callback) => { 
-        // Assignment: how do we register someone?  
+       
         try{
             const registerResponse  = await axios.post(
                 'http://localhost:8000/auth/register', 
                 { email: email, password: password }, 
                 { 'content-type': 'application/json' }
             )  
-            const decoded = jwt(registerResponse.data.token)
-            setAuth({ token: registerResponse.data.token, user: decoded.user })
             callback()       
         } catch (err){
             console.log(`Registering Error ${err}`)
+            navigate('/register')
```

Comment:
This doesn't do what I think you're hoping it does. Refer to the `useNavigation` hook
https://reactrouter.com/docs/en/v6/getting-started/overview#navigation